enumerable.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = data_type if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = check_data_type(self)
+    for i in arr
+      yield(i)
+    end
+    arr
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = check_data_type(self)
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    arr
+  end
+
+  ## my_select method
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any?
+    unless block_given?
+      for i in self
+        return true if i == true
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return false if i == false
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_any?)
+    end
+    bul = false
+    arr = self
+    arr.my_each { |x| bul = true if yield(x) }
+    bul
+  end
+
+  ## my_all method
+  # rubocop: disable Metrics/CyclomaticComplexity
+  def my_all?
+    unless block_given?
+      for i in self
+        return false if i == false || i.nil?
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return true if i == true
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_all?)
+    end
+    bul = false
+    arr = self
+    arr.my_each { |x| bul = true if yield(x) }
+    bul
+  end
+  # rubocop: enable Metrics/CyclomaticComplexity
+
+  ## my_none method
+  def my_none?
+    unless block_given?
+      for i in self
+        return false if i == true
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return true if i == false
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_none?)
+    end
+    bul = true
+    arr = self
+    arr.my_each { |x| bul = false if yield(x) }
+    bul
+  end
+
+  ## my_count method
+  def my_count(args = nil)
+    counter = 0
+    unless args.nil?
+      for i in self
+        counter += 1 if i == args
+       end
+      return counter
+    end
+    counter
+  end
+
+  ## my_map method
+  def my_map(&factor)
+    return to_enum(:my_map) unless block_given?
+
+    arr = self
+    new_arr = []
+    if factor.nil?
+      arr.my_each { |x| new_arr << x if factor.call(x) }
+      arr
+    else
+      arr.my_each { |x| new_arr << x if yield(x) }
+      new_arr
+    end
+  end
```

Comment:
- [ ] #my_map returns a new array with the results of running a given block once for every element in enum.
      Failure/Error: expect(array.my_map(&block)).to eq array.map(&block)      
        expected: [false, true, true, true, true, true, true, true, true, false, true, true, true, true, false, false, ..., true, true, true, true, false, false, false, false, false, true, false, false, false, true, false]
             got: [0, 1, 1, 2, 2, 0, 0, 0, 3, 2, 3, 1, 3, 1, 2, 3, 2, 2, 3, 2, 2, 1, 2, 2, 0, 2, 2, 0, 3, 2, 0, 0, 3, 3, 2, 0, 0, 2, 3, 3, 2, 3, 3, 2, 1, 0, 3, 1, 2, 1, 3, 1]. You need to refactor the code to account for this.

- [ ] #my_map returns a new array with the results of running a given block once for every element in enum::range
      Failure/Error: expect(range.my_map(&block)).to eq range.map(&block)      
        expected: [false, false, false, false, false, false, false, false, false, false, false, false, false, false, fa...e, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
             got: []

- [ ] #my_map accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: expect(array.my_map(&my_proc)).to eq array.map(&my_proc)      
        expected: [false, false, false, false, false, false, false, false, false, false, false, false, false, false, fa...e, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
             got: []

- [ ] #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)
`wrong number of arguments (given 1, expected 0)`